homeassistant/components/soundtouch/media_player.py

```diff
@@ -94,6 +94,23 @@
 )
 
 
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Load Bose Soundtouch media player based on a config entry."""
+    if DATA_SOUNDTOUCH not in hass.data:
+        hass.data[DATA_SOUNDTOUCH] = []
+
+    name = config_entry.data[CONF_NAME]
+    host = config_entry.data[CONF_HOST]
+    port = config_entry.data[CONF_PORT]
+    config = {"id": "ha.component.soundtouch", "host": host, "port": port}
+
+    bose_soundtouch_entity = await hass.async_add_executor_job(
+        SoundTouchDevice, name, config
+    )
+    hass.data[DATA_SOUNDTOUCH].append(bose_soundtouch_entity)
+    async_add_entities([bose_soundtouch_entity])
+
+
 def setup_platform(hass, config, add_entities, discovery_info=None):
```

Comment:
Switch this to `async_setup_platform`, and import the existing config into a new import config flow with SOURCE_IMPORT